gocov/test.go

```diff
@@ -21,27 +21,93 @@
 package main
 
 import (
+	"bytes"
+	"fmt"
 	"io/ioutil"
+	"log"
 	"os"
 	"os/exec"
+	"path/filepath"
+	"strings"
 )
 
+// resolvePackages returns a slice of resolved package names, given a slice of
+// package names that could be relative or recursive.
+func resolvePackages(pkgs []string) ([]string, error) {
+	var buf bytes.Buffer
+	cmd := exec.Command("go", append([]string{"list", "-e"}, pkgs...)...)
+	cmd.Stdin = os.Stdin
+	cmd.Stdout = &buf
+	cmd.Stderr = os.Stderr
+	err := cmd.Run()
+	if err != nil {
+		return nil, err
+	}
+	var resolvedPkgs []string
+	lines := strings.Split(buf.String(), "\n")
+	for _, line := range lines {
+		line = strings.TrimSpace(line)
+		if len(line) > 0 {
+			resolvedPkgs = append(resolvedPkgs, line)
+		}
+	}
+	return resolvedPkgs, nil
+}
+
+func splitPkgsFlags(args []string) ([]string, []string) {
+	var pkgs, flags []string
+	for _, arg := range args {
+		if len(arg) > 0 && arg[0] == '-' {
```

Comment:
This does mean you *have* to use "-flag=value" and never "-flag value", and if you don't you're going to get weird errors... but I think that's fine given the alternatives.